### PR TITLE
DUGK-66 raised threshold of similar-code to 45 (default for TS)

### PR DESCRIPTION
**Pull request by SyntaxXeror**

### CODE REVIEW

**Comment by Omarley7** [comment]
Definitely better than last time! But there are a few issues, that appear when analyzing the .yml file. 

```
WARNING: unknown engine or channel <checkstyle:beta>
WARNING: 'engines' has been deprecated, please use 'plugins' instead
WARNING: 'exclude_paths' has been deprecated, please use 'exclude_patterns' instead
WARNING: 'ratings' has been deprecated, and will not be used
```

Is it possible to squash these errors? When I run [codeclimate CLI](https://github.com/codeclimate/codeclimate) with the .yml file, it takes forever. Possibly because of errors. Don't know why it's working on Github Actions. Perhaps we are using a different version somehow?

Is version number required in the .yml file? Or would it help?

**Review by Omarley7 — CHANGES_REQUESTED** [review]
Better, but perhaps we can get rid of the error posted in prev comment?

**Comment by Omarley7** [comment]
Also, I think the configuration file is running on the old version, which makes the current documentation ridiculously difficult to trust. I think [moving to V2](https://github.com/codeclimate/codeclimate-duplication/issues/120#issuecomment-675696613) makes a lot of sense.

**Comment by SyntaxXeror** [comment]
I basically just updated the version now, and now we don't have to specify mass-threshold, it is just automatic on the defaults in "duplication: config: languages:"

File: .codeclimate.yml [hunk 1]
```diff
@@ -1,42 +1,26 @@
...
-    - "**.module"
-    - "**.sh"
-    - "**.bash"
-exclude_paths:
+exclude_patterns:
   - "**.md"
   - "**.txt"
   - "**.json"
   - docs/
-  - .husky/
```
Comment (Omarley7) [inline .codeclimate.yml:42]: Please add these directories
```yml
  - '**/node_modules/'
  - '**/build/'
  - '**/public/'
```

File: .codeclimate.yml [hunk 2]
```diff
@@ -1,42 +1,26 @@
...
-        - typescript
-        - javascript
-        - python
-        - php
+        - typescript:
+        - javascript:
+        - python:
+        - php:
   eslint:
     enabled: true
```
Comment (Omarley7) [inline .codeclimate.yml:17]: We need to add:
```yml
channel: "eslint-8"
```
See [docs](https://docs.codeclimate.com/docs/eslint#eslint-versions)

File: .codeclimate.yml [hunk 3]
```diff
@@ -1,42 +1,26 @@
...
-    channel: "beta"
   csslint:
     enabled: true
   duplication:
     enabled: true
     config:
-      enabled: true
-      threshold: 7
       languages:
-        - typescript
```
Comment (Omarley7) [inline .codeclimate.yml:16]: Thanks to the updated yml file, it was relatively quick for me to test how many duplications we would get. Here are my findings: 
- current: 20
- 45 (default): 18
- 57: 13

Just ignore this comment. When I started writing, I thought there was no change. And at this point, I didn't want to delete it.

**Review by Omarley7 — CHANGES_REQUESTED** [review]
Much better. It actually runs, even super quickly. I strongly advise installing the CodeClimate CLI on WSL. I'll try tomorrow myself and send a guide if successful. Should be relatively painless.

When the last two things have been added, we are good to go.

File: .codeclimate.yml [hunk 4]
```diff
@@ -1,42 +1,30 @@
...
-        - php
+        - typescript:
+        - javascript:
+        - python:
+        - php:
   eslint:
     enabled: true
+    channel: "eslint-8"
     config:
       config: .eslintrc
```
Comment (SyntaxXeror) [inline .codeclimate.yml]: "Cannot read config file: /code/.eslintrc", nej det er da klart når der ikke ligger en .eslintrc i root ╰（‵□′）╯

Comment (SyntaxXeror) [reply .codeclimate.yml]: (der burde så måske ligge en i root)

Comment (SyntaxXeror) [reply .codeclimate.yml]: For now har jeg sat den til at tage ESlint reglerne fra client/.eslintrc

**Review by Omarley7 — APPROVED** [review]
Superb

File: .codeclimate.yml (hunk 4 above)
Comment (Omarley7) [reply .codeclimate.yml]: Hmm, faktisk meget god pointe... Lad os tage den med til den store PR

Comment (Omarley7) [reply .codeclimate.yml]: Det har også været oppe og vende [her](https://github.com/INTO-CPS-Association/DTaaS/issues/24#issue-1645503049). Jeg hører lige Prasad imorgen om hvorfor den er streget over 🤔

**Pull request merged**